MaxText/layers/nnx_wrappers.py

```diff
@@ -396,36 +383,21 @@ def _module_kwargs():
       out = method_fn(module, *args, **kwargs)
       return out
 
-    # create state
+    # create the nnx module
+    module = self.nnx_class(*self.args, **_module_kwargs())
+    # update nnx module from linen variables
     def maybe_unbox(x):
       if isinstance(x, meta.AxisMetadata):
         return x.unbox()
       return x
-
     states = jtu.tree_map(
       maybe_unbox,
       list(self.variables.values()),
       is_leaf=lambda x: isinstance(x, meta.AxisMetadata),
     )
     if not states:
       states = ({},)
-
-    # update module state
-    if self.abstract_init:
-      module = nnx.eval_shape(lambda: self.nnx_class(*self.args, **_module_kwargs()))
-    else:
-      module = self.nnx_class(*self.args, **_module_kwargs())
```

Comment:
Just for my understanding - could you explain why removing the eval_shape means we don't need to pass in abstract_init=False anymore?

Reply:
correct!